### DISCUSSION TITLE
[Feature]: Authentication Dialogue [Biometric/Pattern/Passcode Unlock]

### DISCUSSION THREAD

**Discussion by EduardDurech** [termux/termux-app#2922] [Ideas]
As per closed request #199, would it still be inhibitively interfering to support a native authentication dialogue? I currently use my phone's AppLock (or Tasker), but they are somewhat insufficient as AppLock will not lock `com.termux/.filepicker.TermuxDocumentsProvider` (as is natively locked in my cloud storages), and Tasker has its own problems. Of course, I could also lock each intent but I feel a native solution would be more appropriate. Obviously, it is important to protect files modifying the system

<img src="https://user-images.githubusercontent.com/39579228/183220937-95bb557f-fe51-427f-9e4a-aa6ef7955ed6.png" width=49%> <img src="https://user-images.githubusercontent.com/39579228/183221706-c38f62e8-4beb-4f11-affc-fc6d79c3d3f0.png" width=49%>

**Comment by sylirre** [comment]
Sorry for late reply, but this is never going to be implemented.

Termux always provided a way for workarounding even total terminal lock down for letting file access & environment recovery: https://wiki.termux.com/wiki/Recover_a_broken_environment
